fixed about:config warning bool

RaafatTurki opened the pull request:
After using the thing I noticed that I'm still getting the `about:config` warning even though I have `general.warnOnAboutConfig` set to `false` in `prefs.js`, after a bit of search I figured that the setting is called `browser.aboutConfig.showWarning` on my firefox (82.0.2 64-bit).

I'll assume this worked on older versions of FF, I don't know if you'd want to set the right one depending on the version or maybe set them both to false in that way feel free to edit my PR.

allo- commented:
Firefox 71 changed this: https://bugzilla.mozilla.org/show_bug.cgi?id=1601899

I wonder if some versions automatically migrated the setting. I do not remember getting the warning again in my profiles.

RaafatTurki commented:
Further research clarified that `general.warnOnAboutConfig` is for the xul version (FF 71-72) and `browser.aboutConfig.showWarning` is for the xhtml version (FF 73+) (the one I added)

both can be reached via the links below depending on your FF version
`FF 71-72` -> `chrome://global/content/config.xul`
`FF 73+` -> `chrome://global/content/config.xhtml`

The Bugzilla report states that it indeed is the HTML version that they're migrating to, However that means my PR effectively has broken this feature on FF 71-72 and fixed it on FF73+ hence why I'm reopening this PR.

The best solution IMO would be to simply enable them both.

allo- commented:
Yes and no.
I am not really a fan of Mozilla changing such things all the time (and the new about:config is worse than the old one), but I think one cannot say the site generates more secure profiles and still target outdated Firefox versions.

Keeping such settings does not harm, but on the other hand is the site meant to keep the generated config short and understandable (in contrast to the many "paste this into your prefs.js to be more secure" lists). This means removing no longer supported options when we notice them is probably a good thing.
